game.exs

```diff
@@ -23,14 +27,13 @@ defmodule GuessingGame do
 
   def play(), do: main(Enum.random(1..100), 7)
 
+  def main(_, 0), do: print_turns(0)
   def main(number, turns) do
```

Comment:
If you had print_outcome return the original outcome (atom) it received, you could probably pipe it all

```elixir
  def main(number, turns) do
    print_turns(turns)
    get_guess() 
    |> compare(number)
    |> print_outcome()
    |> case do
        :equal -> nil
        _ -> main(number, turns-1)
    end
  end
```